cocos/core/math/mat4.ts

```diff
@@ -1985,6 +1985,26 @@ export class Mat4 extends ValueType {
         return this;
     }
 
+    /**
+     * @zh Set translation elements.
+     * @param translation position
+     */
+    public setTranslation (translation: Vec3) {
+        this.m03 = translation.x;
+        this.m07 = translation.y;
+        this.m11 = translation.z;
+    }
+
+    /**
+     * @zh Set scaling elements.
+     * @param scale scale
+     */
+    public setScale (scale: Vec3) {
```

Comment:
Could you please comment out the various cases? For example, what if the matrix invoking this method was not identified and what if it'd already had a rotation part?

Reply:
Setting the scaling process is irreversible in matrix operations, please use it with caution

Reply:
> Setting the scaling process is irreversible in matrix operations, please use it with caution

I mean what if I call this method on an identified matrix:
```ts
new Mat4().setScale(new Vec3(2, 2, 2));
```
What if I call it with a matrix that has already had a rotation part?
```ts
const m = new Mat4();
m.fromRTS(Quat.fromAxisAngle(new Quat(), new Vec3(), Math.PI), new Vec3(), new Vec3(1, 1, 1));
m.setScale(new Vec3(2, 2, 2)); // Would it hurt my rotation?
```

Please detail its functionality in comment, and you'd better explicitly explain to developers how this function affects any part of the matrix.

Reply:
yes， it is very necessary